zebra-chain/src/parameters/network.rs

```diff
@@ -279,3 +275,84 @@ impl FromStr for Network {
+            zp_consensus::NetworkUpgrade::Sapling => NetworkUpgrade::Sapling,
+            zp_consensus::NetworkUpgrade::Blossom => NetworkUpgrade::Blossom,
+            zp_consensus::NetworkUpgrade::Heartwood => NetworkUpgrade::Heartwood,
+            zp_consensus::NetworkUpgrade::Canopy => NetworkUpgrade::Canopy,
+            zp_consensus::NetworkUpgrade::Nu5 => NetworkUpgrade::Nu5,
+        };
+
+        // Heights are hard-coded below Height::MAX or checked when the config is parsed.
+        target_nu
+            .activation_height(self)
+            .map(|Height(h)| zp_consensus::BlockHeight::from_u32(h))
+    }
+
+    fn coin_type(&self) -> u32 {
+        match self {
+            Network::Mainnet => zp_constants::mainnet::COIN_TYPE,
+            // The regtest cointype reuses the testnet cointype,
+            // See <https://github.com/satoshilabs/slips/blob/master/slip-0044.md>
+            Network::Testnet(_) => zp_constants::testnet::COIN_TYPE,
+        }
+    }
+
+    fn address_network(&self) -> Option<zcash_address::Network> {
+        match self {
+            Network::Mainnet => Some(zcash_address::Network::Main),
+            Network::Testnet(params) if params.is_default_testnet() => {
+                Some(zcash_address::Network::Test)
+            }
+            _other => None,
+        }
+    }
+
+    fn hrp_sapling_extended_spending_key(&self) -> &str {
+        match self {
+            Network::Mainnet => zp_constants::mainnet::HRP_SAPLING_EXTENDED_SPENDING_KEY,
+            Network::Testnet(params) if params.is_default_testnet() => {
+                zp_constants::testnet::HRP_SAPLING_EXTENDED_SPENDING_KEY
+            }
+            // TODO: Add this as a field to testnet params
+            _other => "secret-extended-key-unknown-test",
+        }
+    }
+
+    fn hrp_sapling_extended_full_viewing_key(&self) -> &str {
+        match self {
+            Network::Mainnet => zp_constants::mainnet::HRP_SAPLING_EXTENDED_FULL_VIEWING_KEY,
+            Network::Testnet(params) if params.is_default_testnet() => {
+                zp_constants::testnet::HRP_SAPLING_EXTENDED_FULL_VIEWING_KEY
+            }
+            // TODO: Add this as a field to testnet params
+            _other => "zxviewunknowntestsapling",
+        }
+    }
+
+    fn hrp_sapling_payment_address(&self) -> &str {
+        match self {
+            Network::Mainnet => zp_constants::mainnet::HRP_SAPLING_PAYMENT_ADDRESS,
+            Network::Testnet(params) if params.is_default_testnet() => {
+                zp_constants::testnet::HRP_SAPLING_PAYMENT_ADDRESS
+            }
+            // TODO: Add this as a field to testnet params
+            _other => "zunknowntestsapling",
+        }
+    }
+
+    fn b58_pubkey_address_prefix(&self) -> [u8; 2] {
+        self.kind().b58_pubkey_address_prefix()
+    }
+
+    fn b58_script_address_prefix(&self) -> [u8; 2] {
+        self.kind().b58_script_address_prefix()
+    }
```

Comment:
Do we want to return these constants directly if this is called very frequently, or do we have trust that rustc will optimize this?